### DISCUSSION TITLE
Is it possible to use the EventSubscriber and know when changes were committed?

### DISCUSSION THREAD

**Discussion by jbmikk** [mikro-orm/mikro-orm#2645] [Q&A]
I am using the event subscriber, specifically the `afterFlush` event to serialize changes to a log system.
I would like to log the changes that were actually committed to the database, so I was looking for something like `afterCommit` to make sure the transaction was successful.
Is there any other event in the system I can listen to to achieve this?
It would be fine if I have to combine the `afterFlush` with another event as long as I can identify the flush event and the commit event belong to the same transaction.

**Comment by B4nan** [comment]
Looks like its not documented in the events page, but we have this one: [afterTransactionCommit ](https://mikro-orm.io/docs/api/enums/core.eventtype#aftertransactioncommit)

> It would be fine if I have to combine the afterFlush with another event as long as I can identify the flush event and the commit event belong to the same transaction.

You will get the `em` and `uow` instances in arguments, you could map it based on that.
